### PR TITLE
changing vtables and accessors to be fileprivate

### PR DESCRIPTION
**Pull request by stephen-hawley**
Change vtable accessors to be `fileprivate` fixing issue [464](https://github.com/xamarin/binding-tools-for-swift/issues/464)

There are three parts to this:
1. Adding `fileprivate` support to Dynamo
2. Changing internal accessors/declarations in OverrideBuilder to fileprivate
3. Minor adjustment to generic grammar that was failing on `where T == Some.Type` (which was blocking tests)

Tests pass.

### CODE REVIEW

File: swiftinterfaceparser/SwiftInterface.g4 [hunk 1]
```diff
@@ -496,7 +496,7 @@ requirement_list : requirement (OpComma requirement)*  ;
 requirement : conformance_requirement | same_type_requirement  ;
 
 conformance_requirement : type_identifier ':' type_identifier | type_identifier ':' protocol_composition_type  ;
-same_type_requirement : type_identifier OpEqEq type  ;
+same_type_requirement : type_identifier operator type  ;
```
Comment (spouliot) [inline swiftinterfaceparser/SwiftInterface.g4:499]: is this related to the PR ?

**Review by spouliot — APPROVED** [review]

File: swiftinterfaceparser/SwiftInterface.g4 (hunk 1 above)
Comment (stephen-hawley) [reply swiftinterfaceparser/SwiftInterface.g4:499]: Only in that I got hip deep into the PR and would have failed all the unit tests that were unignored (see #3 above). I was tempted to put it into a separate PR, but it was a single word change...
